quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -71,23 +76,55 @@ impl LocalFileStorage {
     /// Moves a file from a source to a destination.
     /// from here is an external path, and to is an internal path.
     pub async fn move_into(&self, from_external: &Path, to: &Path) -> crate::StorageResult<()> {
-        let to_full_path = self.root.join(to);
+        let to_full_path = self.full_path(to)?;
         fs::rename(from_external, to_full_path).await?;
         Ok(())
     }
 
     /// Moves a file from a source to a destination.
     /// from here is an internal path, and to is an external path.
     pub async fn move_out(&self, from_internal: &Path, to: &Path) -> crate::StorageResult<()> {
-        let from_full_path = self.root.join(from_internal);
+        let from_full_path = self.full_path(from_internal)?;
         fs::rename(from_full_path, to).await?;
         Ok(())
     }
+
+    async fn delete_single_file(&self, relative_path: &Path) -> StorageResult<()> {
+        let full_path = self.full_path(relative_path)?;
+        ignore_error_kind!(ErrorKind::NotFound, fs::remove_file(full_path).await)?;
+        Ok(())
+    }
+}
+
+/// Ensure that the path given does not include any ".." for security reasons.
+///
+/// In order to reduce the attack surface, we want to make sure the `FileStorage`
+/// only access/delete files that are children of its root_directory.
+fn ensure_no_parent_dir_double_dot(path: &Path) -> StorageResult<()> {
```

Comment:
I think this could be called something like `ensure_valid_path` or similar as this does technically do more than just ensure the parent isn't a `..` for example, `/todo` will also get rejected by this routine.

Reply:
If you just have "nitpicks" comment, please mark the CR as approved. This is a practise that is a bit specific to quickwit: we are all in different timezone, so it is critical that we keep the process asynchronous when possible.

Reply:
(I did the rename and will abuse my privilege and merge that  PR  :) )

Reply:
Ah apologies 😅

Reply:
No problem! you could not know! :)